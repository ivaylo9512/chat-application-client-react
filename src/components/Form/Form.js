import React, { useEffect } from 'react'
import useInput from '../../hooks/useInput'
import './Form.css'
import { useDispatch, useSelector } from 'react-redux'
import { getSearchVisibility } from '../../app/slices/stylesSlice'

const Form = ({action, resetState, selector,  placeholder, onUnmount}) => {
    const [inputValue, input] = useInput({type:'text', placeholder})
    const isSearchHidden = useSelector(getSearchVisibility);
    const dispatch = useDispatch();
    const query = useSelector(selector);

    const submit = (e) => {
        e.preventDefault()
        dispatch(resetState());
        dispatch(action({...query, name: inputValue, pages: 1}));
    }

    useEffect(() => {
        if(onUnmount){
            return () => dispatch(onUnmount())
        }
    }, [])

    return (
        <div className={`form-container${isSearchHidden ? ' hidden' : ''}`}>
            <form onSubmit={submit}>
                {input}
                <button><i className='fas fa-search'></i></button>
            </form>
        </div>
    )
}
export default Form