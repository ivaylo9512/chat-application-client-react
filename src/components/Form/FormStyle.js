import styled from 'styled-components'

export const Container = styled.div`
    & .second-style{
        height: 5vw;
        width: 5vw;
        position: absolute;
        top: 50%;
        left: 4%;
        right: unset;
        bottom: unset;
        transform: translateY(-250%);
    }
`
export const Button = styled.button`
    width: 9vw;
    background: #181b21;
    border: 0;
    color: white;
    outline: none;
    font-size: 1vw;
    transition: 1.5s;
    border-radius: 5vw;
    cursor: pointer;
    border-radius: ${props => props.isHidden 
        ? '0 5vw 5vw 0'
        : '5vw 5vw 5vw 5vw'};
`
export const FormNode = styled.form`
    display: flex;
    position: absolute;
    justify-content: flex-end;
    background: hsl(220deg 16% 11%);
    color: white;
    font-size: 1vw;
    transition: 1.5s;
    cursor: pointer;
    box-shadow: 0px 0px 0.3vw 0.1vw rgb(0 0 0 / 65%);
    bottom: 3vw;
    right: 4%;
    transition: transform 2s 1.75s,width 1.5s,height 1.5s,border-radius 1.5s;

    height: ${props => props.isHidden ? '11vh' : '4vw'};
    width: ${props => props.isHidden ? '40vw' : '4vw'};
    
    border-radius: ${props => {
        if(props.isHidden) return '0.5vw 5vh 5vh 0.5vw';

        switch(1){
            case 1:
                return '0.5vw 5vh 5vh 5vh';
            case 2: 
                return '0.5vh 3vh 3vh 3vh';
            }
        }
    };

    input::placeholder {
        color: white;
        opacity: 1;
        font-size: 0.8vw;
        font-weight: bold;
    }
      
    input:-ms-input-placeholder {
        color: white;
        font-size: 0.8vw;
        font-weight: bold;
    }
      
    input::-ms-input-placeholder {
        color: white;
        font-size: 0.8vw;
        font-weight: bold;
    }
    
    input{
        flex: 1;
        border-radius: 0.5vw 0px 0px 0.5vw;
        text-align: center;
        outline: none;
        box-sizing: border-box;
        font-size: 1vw;
        background: #181b21;
        color: white;
        width: 0;
        padding: 0;
        border: 0;
        position: relative;
        right: -0.2vw;        
        max-width: ${props => props.isHidden ? 'none' : 0};
    }
`