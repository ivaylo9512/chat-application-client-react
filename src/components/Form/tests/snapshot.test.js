import { shallow } from 'enzyme';
import Form from 'components/Form/Form';
import * as Redux from 'react-redux';

describe('Pagination snapshot tests', () => {
    beforeEach(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());
    })
    const createWrapper = (state) => {
        jest.spyOn(Redux, 'useSelector').mockReturnValue(state);

        return shallow(<Form />)   
    }

    it('should call dispatch on submit', () => {})
    it('should render snanpshot', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should render snanpshot with false', () => {
        const wrapper = createWrapper(false);

        expect(wrapper).toMatchSnapshot();
    })
});