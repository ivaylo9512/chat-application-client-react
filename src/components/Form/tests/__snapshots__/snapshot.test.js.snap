// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot tests should render snanpshot 1`] = `
<styled.div
  isHidden={true}
>
  <styled.form
    className="second-style"
    isHidden={true}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <styled.button
      isHidden={true}
    >
      <i
        className="fas fa-search"
      />
    </styled.button>
  </styled.form>
</styled.div>
`;

exports[`Form snapshot tests should render snanpshot with false 1`] = `
<styled.div
  isHidden={false}
>
  <styled.form
    className="second-style"
    isHidden={false}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <styled.button
      isHidden={false}
    >
      <i
        className="fas fa-search"
      />
    </styled.button>
  </styled.form>
</styled.div>
`;
